src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -293,23 +297,35 @@ private void OnWriteCompleted(WriteContext writeContext)
             {
                 PoolWriteContext(writeContext);
 
-                // _numBytesPreCompleted can temporarily go negative in the event there are
-                // completed writes that we haven't triggered callbacks for yet.
-                _numBytesPreCompleted -= bytesWritten;
-
-                // bytesLeftToBuffer can be greater than _maxBytesPreCompleted
-                // This allows large writes to complete once they've actually finished.
-                var bytesLeftToBuffer = _maxBytesPreCompleted - _numBytesPreCompleted;
-                while (_tasksPending.Count > 0 &&
-                       (int)(_tasksPending.Peek().Task.AsyncState) <= bytesLeftToBuffer)
+                if (_lastWriteError == null)
                 {
-                    var tcs = _tasksPending.Dequeue();
-                    var bytesToWrite = (int)tcs.Task.AsyncState;
+                    // _numBytesPreCompleted can temporarily go negative in the event there are
+                    // completed writes that we haven't triggered callbacks for yet.
+                    _numBytesPreCompleted -= bytesWritten;
+
+                    // bytesLeftToBuffer can be greater than _maxBytesPreCompleted
+                    // This allows large writes to complete once they've actually finished.
+                    var bytesLeftToBuffer = _maxBytesPreCompleted - _numBytesPreCompleted;
+                    while (_tasksPending.Count > 0 &&
+                           (int)(_tasksPending.Peek().Task.AsyncState) <= bytesLeftToBuffer)
+                    {
+                        var tcs = _tasksPending.Dequeue();
+                        var bytesToWrite = (int)tcs.Task.AsyncState;
 
-                    _numBytesPreCompleted += bytesToWrite;
-                    bytesLeftToBuffer -= bytesToWrite;
+                        _numBytesPreCompleted += bytesToWrite;
+                        bytesLeftToBuffer -= bytesToWrite;
 
-                    _tasksCompleted.Enqueue(tcs);
+                        _tasksCompleted.Enqueue(tcs);
+                    }
+                }
+                else
+                {
+                    // Error all waiting tasks
+                    while (_tasksPending.Count > 0)
```

Comment:
We decided not to add an `IgnoreWriteExceptions` property. We will never throw from `SocketOutput.WriteAsync`. `FrameResponseStream` will be modified to honor the `CancellationToken` that is passed in for async writes.

If you want the `IgnoreWriteExceptions = false` behavior you will always pass in the `IHttpRequestLifetimeFeature.RequestAborted` token to your writes. If `FrameResponseStream` ever throws due to a canceled token it will abort the connection.

We will also plan to log all write failures using `LogLevel.Debug` correlated with the connection id. The connection id will be made accessible from user code via a new `IHttpConnectionFeature.ConnectionId` property.